diff --git a/smb.asm b/smb.asm
index e5ef1ad..c6d368c 100644
--- a/smb.asm
+++ b/smb.asm
@@ -2717,8 +2717,8 @@ DoneInitArea:  lda #Silence             ;silence music
 PrimaryGameSetup:
       lda #$01
       sta FetchNewGameTimerFlag   ;set flag to load game timer from header
-      sta PlayerSize              ;set player's size to small
       lda #$02
+      sta PlayerSize              ;set player's size to small
       sta NumberofLives           ;give each player three lives
       sta OffScr_NumberofLives
 
