diff --git a/smb.asm b/smb.asm
index e5ef1ad..e12c81a 100644
--- a/smb.asm
+++ b/smb.asm
@@ -985,8 +985,10 @@ ChkSelect:    cmp #Select_Button          ;check to see if the select button was
               jsr DemoEngine              ;run through the demo actions
               bcs ResetTitle              ;if carry flag set, demo over, thus branch
               jmp RunDemo                 ;otherwise, run game engine for demo
-ChkWorldSel:  ldx WorldSelectEnableFlag   ;check to see if world selection has been enabled
-              beq NullJoypad
+ChkWorldSel:  ldx #$00                    ;check to see if world selection has been enabled
+              nop
+              nop
+              nop
               cmp #B_Button               ;if so, check to see if the B button was pressed
               bne NullJoypad
               iny                         ;if so, increment Y and execute same code as select
@@ -1008,7 +1010,8 @@ SelectBLogic: lda DemoTimer               ;if select or B pressed, check demo ti
 IncWorldSel:  ldx WorldSelectNumber       ;increment world select number
               inx
               txa
-              and #%00000111              ;mask out higher bits
+              nop
+              nop
               sta WorldSelectNumber       ;store as current world select number
               jsr GoContinue
 UpdateShroom: lda WSelectBufferTemplate,x ;write template for world select in vram buffer
