diff --git a/smb.asm b/smb.asm
index e5ef1ad..4d4502a 100644
--- a/smb.asm
+++ b/smb.asm
@@ -986,7 +986,10 @@ ChkSelect:    cmp #Select_Button          ;check to see if the select button was
               bcs ResetTitle              ;if carry flag set, demo over, thus branch
               jmp RunDemo                 ;otherwise, run game engine for demo
 ChkWorldSel:  ldx WorldSelectEnableFlag   ;check to see if world selection has been enabled
-              beq NullJoypad
+;              beq NullJoypad
+              nop
+              nop
+              nop
               cmp #B_Button               ;if so, check to see if the B button was pressed
               bne NullJoypad
               iny                         ;if so, increment Y and execute same code as select
@@ -2539,7 +2542,7 @@ PrintStatusBarNumbers:
 OutputNumbers:
              clc                      ;add 1 to low nybble
              adc #$01
-             and #%00001111           ;mask out high nybble
+             and #%11111111           ;mask out high nybble
              cmp #$06
              bcs ExitOutputN
              pha                      ;save incremented value to stack for now and
