diff --git a/smb.asm b/smb.asm
index e5ef1ad..e96b93a 100644
--- a/smb.asm
+++ b/smb.asm
@@ -12333,7 +12333,8 @@ RImpd: ldx #$02                  ;return $02 to X
 NXSpd: ldy #$10
        sty SideCollisionTimer    ;set timer of some sort
        ldy #$00
-       sty Player_X_Speed        ;nullify player's horizontal speed
+       nop
+       nop
        cmp #$00                  ;if value set in A not set to $ff,
        bpl PlatF                 ;branch ahead, do not decrement Y
        dey                       ;otherwise decrement Y now
