diff --git a/smb.asm b/smb.asm
index b149bdc..d302a69 100644
--- a/smb.asm
+++ b/smb.asm
@@ -867,12 +867,11 @@ ChkStart:      lda SavedJoypad1Bits   ;check to see if start is pressed
                lda GamePauseStatus    ;check to see if timer flag is set
                and #%10000000         ;and if so, do not reset timer (residual,
                bne ExitPause          ;joypad reading routine makes this unnecessary)
-               lda #$2b               ;set pause timer
-               sta GamePauseTimer
-               lda GamePauseStatus
+               lda PlayerSize         ;set pause timer
+               eor #$01
+               sta PlayerSize
+               lda #$01
                tay
-               iny                    ;set pause sfx queue for next pause mode
-               sty PauseSoundQueue
                eor #%00000001         ;invert d0 and set d7
                ora #%10000000
                bne SetPause           ;unconditional branch
